autoload/go/template.vim

```diff
@@ -5,6 +5,9 @@ set cpo&vim
 let s:current_file = expand("<sfile>")
 
 function! go#template#create() abort
+  if !&modifiable
+    return
+  endif
```

Comment:
Given that this function does not write the file, why is this change necessary?

Reply:
Added some clarification on related issue; but maybe the check should be somewhere else in the call stacK?

Reply:
Instead of returning early here, it's may be better to return early in https://github.com/fatih/vim-go/blob/master/plugin/go.vim#L278-L283

Reply:
Done. Opted for an additional check instead of early return since I felt it was still clear given the var name, but please let me know if you'd still prefer it as an early return.